base/strings/util.jl

```diff
@@ -464,11 +464,89 @@ function hex2bytes(s::AbstractString)
 end
 
 """
+    hex2bytes(s::AbstractVector{UInt8})
+
+Convert the hexadecimal bytes array to its binary representation. Returns
+`Vector{UInt8}`, i.e. a vector of bytes.
+"""
+@inline function hex2bytes(s::AbstractVector{UInt8})
+    len = length(s)
+    if isodd(len)
+        throw(ArgumentError("source vector length must be even"))
+    end
+    d = zeros(UInt8, div(len, 2))
```

Comment:
Why initialize to zero?  Just use `Vector{UInt8}(len >> 1)` to allocate an uninitialized array.